src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/LongPollingTransport.java

```diff
@@ -75,18 +73,27 @@ public Completable start(String url) {
                     this.active = true;
                 }
                 this.threadPool = Executors.newCachedThreadPool();
-                threadPool.execute(() -> poll(url).subscribeWith(receiveLoop));
+                threadPool.execute(() -> {
+                    this.onReceiveThread = Executors.newSingleThreadExecutor();
+                    receiveLoop.subscribe(() -> {
+                        this.stop().onErrorComplete().subscribe();
+                    }, e -> {
+                        this.stop().onErrorComplete().subscribe();
+                    });
+                    poll(url)
+                    .subscribeWith(receiveLoop);
+                });
 
                 return Completable.complete();
             });
-        });
+        }));
     }
 
     private Completable poll(String url) {
         if (this.active) {
             pollUrl = url + "&_=" + System.currentTimeMillis();
             logger.debug("Polling {}.", pollUrl);
-            return this.updateHeaderToken().flatMapCompletable((x) -> {
+            return this.updateHeaderToken().andThen(Completable.defer(() -> {
```

Comment:
What's the difference between `.andThen(Completable.defer(() ->` and `.flatMapCompletable((x) ->`. Just the lack of the unused `x` argument?

Reply:
Right, and updateHeaderToken no longer returns a value.